nautobot/ipam/mixins.py

```diff
@@ -0,0 +1,27 @@
+class LocationFieldReplaceMixin:
+    """
+    A mixin for Django QuerySets to support backward compatibility by converting
+    queries from a previously used 'location' field to the new
+    'locations'. This mixin intercepts `filter` and `exclude` calls
+    to transform references from 'location' to 'locations'.
+    """
+
+    def _convert_location_to_locations(self, kwargs):
+        """Transforms query parameters that reference 'location' field into the corresponding 'locations' field."""
+        updated_kwargs = {}
+        for field, value in kwargs.items():
+            if field.startswith("location") and not field.startswith("locations"):
+                _, lookup_expr = field.split("location", maxsplit=1)
+                locations_field = f"locations{lookup_expr}".strip()
```

Comment:
Do we not need to convert `location` to `locations_in` or anything like that?

Definitely we'll want to have some extensive testing added for this.